acme/http_challenge.go

```diff
@@ -95,12 +95,13 @@ Loop:
 func (s *httpChallenge) startHTTPServer(domain string, token string, keyAuth string) {
 
 	// Allow for CLI port override
-	port := ":80"
+	port := domain + ":80"
 	if s.optPort != "" {
 		port = ":" + s.optPort
 	}
```

Comment:
The change is to make it so if you specify a port number it defaults to the fallback behavior; it doesn't ignore it just make it happen up front. It could maybe be optimized to not try the fallback behavior (too), but I'm not sure there's any harm in trying and failing the same listen twice.

The old behavior (listen on the DNS resolved IP) doesn't make sense if you are specifying a port since you need something else to forward the port and who knows what you are trying to do.